rcon/discord_chat.py

```diff
@@ -49,13 +57,182 @@ def parse_webhook_url(url):
-    except Exception as e:
-        logger.error("error parsing Discord chat webhook url: %s", e)
-        return None, None
+        return None
+    resp = requests.get(url).json()
+    _id = int(resp["id"])
+    token = resp["token"]
+    return _id, token
+
+
+def make_hook(webhook_url):
+    webhook_id, webhook_token = parse_webhook_url(webhook_url)
+    return Webhook.partial(
+        id=webhook_id, token=webhook_token, adapter=RequestsWebhookAdapter()
+    )
+
+
+class DiscordWebhookHandler:
+    """RCON Tool -> Discord messaging."""
+
+    def __init__(self):
+        # TODO: take config as argument if modularity is desired.
+        self.ping_trigger_words = []
+        self.ping_trigger_roles = []
+        self.chat_webhook = None
+        self.kills_webhook = None
+        self.ping_trigger_webhook = None
+        self.send_kills = False
+        self.send_team_kills = False
+        self.init_env_vars()
+
+    def init_env_vars(self):
+        self.init_ping_trigger_vars()
+        self.init_chat_vars()
+        self.init_kill_vars()
+
+    def init_ping_trigger_vars(self):
+        try:
+            self.ping_trigger_words = [
+                word.lower().strip()
+                for word in PING_TRIGGER_WORDS.split(",") if word
+            ]
+            self.ping_trigger_roles = [
+                role.strip()
+                for role in PING_TRIGGER_ROLES.split(",") if role
+            ]
+            self.ping_trigger_webhook = make_hook(PING_TRIGGER_WEBHOOK)
+            logger.info("ping trigger variables initialized successfully")
+        except Exception as e:
+            logger.exception("error initializing ping trigger variables: %s", e)
+
+    def init_chat_vars(self):
+        try:
+            self.chat_webhook = make_hook(DISCORD_CHAT_WEBHOOK_URL)
+            logger.info("chat variables initialized successfully")
+        except Exception as e:
+            logger.exception("error initializing chat variables: %s", e)
+
+    def init_kill_vars(self):
+        try:
+            self.kills_webhook = make_hook(DISCORD_KILLS_WEBHOOK_URL)
+            self.send_kills = True if SEND_KILLS == "yes" else False
+            self.send_team_kills = True if SEND_TEAM_KILLS == "yes" else False
+            logger.info("kill variables initialized successfully")
+        except Exception as e:
+            logger.exception("error initializing kill variables: %s", e)
+
+    def send_chat_message(self, _, log):
+        if not self.chat_webhook:
+            return
+
+        try:
+            message = log["sub_content"]
+
+            if ALLOW_MENTIONS != 'yes':
+                message = discord.utils.escape_mentions(message)
+            message = discord.utils.escape_markdown(message)
+
+            player = log["player"].ljust(25)
+            steam_id = log["steam_id_64_1"]
+            action = log["action"]
+            action = action.split("CHAT")[1]
+            color = CHAT_ACTION_TO_COLOR[action]
```

Comment:
If you don't want that to break at the next update (and also when structured_logs fails to parse properly) you should `.get()` or catch the `KeyError` with a default

Reply:
Ah nevermind, it's the line above that could potential break with an `IndexError` actually